chromium_src/third_party/blink/renderer/bindings/scripts/bind_gen/interface.py

```diff
@@ -223,17 +223,31 @@ def _append_report_page_graph_api_call_event(cg_context, expr):
     else:
         return_value = _to_page_graph_value("return_value")
 
-    pattern = (
-        ";\n"
-        "if (UNLIKELY(${page_graph_enabled})) {{\n"
-        "  if (auto pg_scope = ScopedPageGraphCall(); pg_scope.has_value()) {{\n"
-        "    probe::RegisterPageGraphWebAPICallWithResult("
-        "      ${current_execution_context}, \n"
-        "      ${page_graph_binding_name}, {_1}, \n"
-        "      CreatePageGraphValues({_2}), \n"
-        "      {_3}, {_4});\n"
-        "  }}"
-        "}}")
+    # The odd pattern below:
+    #     auto *pg_v8_receiver = ${v8_receiver};
+    # is to make sure the code generator sets up `v8_receiver` correctly
+    # (which necessarily happens before the PageGraph check block),
+    # but to only interact with it in cases where we know there will be
+    # a context associated with it.
+    pattern = (";\n"
+               "if (UNLIKELY(${page_graph_enabled})) {{\n"
+               "  if (auto pg_scope = ScopedPageGraphCall();"
+               "    pg_scope.has_value()) {{\n"
+               "    auto pg_v8_receiver = ${v8_receiver};\n"
+               "    auto receiver_context = pg_v8_receiver->"
+               "        GetCreationContextChecked();\n"
```

Comment:
I saw a failure on CI when you used `${receiver_context}` directly:
```
[1481227:1:0429/222413.714941:FATAL:v8_initializer.cc(798)] V8 error: No creation context available (v8::Object::GetCreationContextChecked).
#0 0x55d3aa118d22 base::debug::CollectStackTrace()
#1 0x55d3aa101973 base::debug::StackTrace::StackTrace()
#2 0x55d3aa000ed9 logging::LogMessage::Flush()
#3 0x55d3aa001a29 logging::LogMessageFatal::~LogMessageFatal()
#4 0x55d3adda5265 blink::ReportV8FatalError()
#5 0x55d3a3eb7c98 v8::Utils::ReportApiFailure()
#6 0x55d3a3edd843 v8::Object::GetCreationContextChecked()
```

which means `GetCreationContextChecked()` can crash and most likely will crash when PG is active.

we can avoid this by using `GetCreationContext()` here and checking `MaybeLocal<>` before using it.

Reply:
i think removing `${receiver_context}` correctly prevents the issue, since if you do `${receiver_context}` then you end up trying to get the receiver context w/o checking for pagegraph (which will fail) but if you do it hte way i did before (which is only ask for the context in a pagegraph case (which will only be webapi or builtin calls) then it is guaranteed to succeed

I think the above is correct.  If not, can you send a fail test case? Just, not sure i follow

Reply:
> i think removing `${receiver_context}` correctly prevents the issue, since if you do `${receiver_context}` then you end up trying to get the receiver context w/o checking for pagegraph (which will fail) but if you do it hte way i did before (which is only ask for the context in a pagegraph case (which will only be webapi or builtin calls) then it is guaranteed to succeed
> 
> I think the above is correct. If not, can you send a fail test case? Just, not sure i follow

well, you saw the browser test failures, right? What you did here is you just moved the crashing code to be active only if PG is enabled.

Do you agree that the failing calls (that break browser tests) can happen in the wild? If they can happen in the wild, then with the enabled PG, the renderer will crash the same.

If you want a test case, just look into what these tests are doing and try to replicate it with PG enabled: https://ci.brave.com/job/pr-brave-browser-correct-frame-id-for-js-calls-linux-x64/10/testReport/

Reply:
apologies but im missing something.  Which browser test failures failed with this version of the PR?

Reply:
> Do you agree that the failing calls (that break browser tests) can happen in the wild?

This is the part i dont understand. I think the change i made will prevent this from happening. But you definitely understand this part of chromium better than i do.  But my understanding is that i fixed the core problem (asking for a receiver context when there isn't a receiver context for the relevant action the probe is reporting), not just that i hid the error

I 100%, definitely could be wrong (like i said, you know chromium far far better than i do), but if the current situation is wrong, i think i need more help to understand why, apologies

Reply:
> apologies but im missing something. Which browser test failures occured with this version of the PR?

try this:
```
npm test browser_tests -- --gtest_filter=AdClickMetricsBrowserTest.SetTopLocationFromCrossOriginAdIframe 
--enable-features=PageGraph
```